source/develop/release-management/features/storage/multipath-events.html.md

```diff
@@ -0,0 +1,167 @@
+---
+title: Multipath Events
+category: feature
+authors: Fred Rolland
+feature_name: Multipath Events
+feature_modules: engine,vdsm
+feature_status: Design
+---
+
+# Multipath Events
+
+
+## Overview
+
+This feature adds the ability to display events to the user in
+the engine UI when multipath device path changes its status.
+The supported events are when a path moves to down or up state.
+
+
+## Owner
+
+- Name: [Fred Rolland](https://github.com/rollandf)
+- Email: <frolland@redhat.com>
+
+
+## Detailed Description
+
+
+### How it works
+
+When a host is connected to a DC (Storage Pool), Vdsm will register
+to udev multipath events. For each event, a message will be sent to Engine
+with the details of the event. Engine will create an audit log event. 
+Once the host is disconnected from the DC, Vdsm will unregister from
+the 'udev' events.
+
+
+### New vdsm API
+
+New multipath event type
+
+
+#### Multipath Event
+
```

Comment:
Event also have a topic, something like ```multipath|status```.

Maybe we should include the multiapth device guid in the event - will allow code to register events to specific multipath device on engine side.

Reply:
guid = WWN/WWPN, I assume?

Reply:
Sure we can. What is the right field ? 
https://www.kernel.org/doc/Documentation/device-mapper/dm-uevent.txt

Reply:
guid is the last path component in the multipath device path /dev/mapper/xxxyyy. We don't call it WWN in vdsm or engine, so better not use this term.